versioned_docs/version-V3/guides/swaps/single-swaps.md

```diff
@@ -4,276 +4,331 @@ title: Single Swaps
+
+Add the following lines to your `contracts/SimpleSwap.sol` file: 
 
 ```solidity
-import '@uniswap/v3-periphery/contracts/interfaces/ISwapRouter.sol';
-import '@uniswap/v3-periphery/contracts/libraries/TransferHelper.sol';
+// ...
+contract SimpleSwap {
+    ISwapRouter public immutable swapRouter;
+    address public constant DAI = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+    address public constant WETH9 = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    uint24 public constant feeTier = 3000;
+		
+	constructor(ISwapRouter _swapRouter) {
+        swapRouter = _swapRouter;
+    }
+}
 ```
 
-Create a contract called `SwapExamples`, and declare an immutable public variable `swapRouter` of type `ISwapRouter`.
-This allows us to call functions in the `ISwapRouter` interface.
+First we use the `ISwapRouter` interface to create a reference to a [SwapRouter](https://docs.uniswap.org/protocol/reference/periphery/interfaces/ISwapRouter). The [SwapRouter](https://docs.uniswap.org/protocol/reference/periphery/interfaces/ISwapRouter) is part of the [Uniswap V3 Periphery](https://github.com/Uniswap/v3-periphery) contracts, designed to make executing swaps easier:  
 
 ```solidity
-contract SwapExamples {
-    // For the scope of these swap examples,
-    // we will detail the design considerations when using `exactInput`, `exactInputSingle`, `exactOutput`, and  `exactOutputSingle`.
-    // It should be noted that for the sake of these examples we pass in the swap router as a constructor argument instead of inheriting it.
-    // More advanced example contracts will detail how to inherit the swap router safely.
-    // This example swaps DAI/WETH9 for single path swaps and DAI/USDC/WETH9 for multi path swaps.
+ISwapRouter public immutable swapRouter;
+```
 
-    ISwapRouter public immutable swapRouter;
+Second, we create constant variables for the ERC20 tokens that we’ll be swapping. You can and should put these addresses into [Ether Scan](https://etherscan.io/) and confirm that they are the tokens you’re expecting: 
+
+```solidity
+address public constant DAI = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+address public constant WETH9 = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
 ```
 
-Hardcode the token contract addresses and pool fee tiers for the example. In production, you would likely use an input parameter for this and pass the input into a memory variable, allowing the contract to change the pools and tokens it interacts with on a per transaction basis, but for conceptual simplicity, we are hardcoding them here.
+Next we create another constant variable to indicate the fee tier of the pool we want to use to swap. Again, in a more general contract this would be an input that gets set at runtime, but for simplicity we’re hardcoding to the 0.3% fee tier pool. Fee tiers are denoted in 1/100ths of a basis point so our fee tier will be 3000: 
```

Comment:
```suggestion
Next we create another constant variable to indicate the fee tier of the pool we want to use to swap. Again, in a more general contract this would be an input that gets set at runtime, but for simplicity we’re hardcoding to the 0.3% fee tier pool. Fee tiers are denoted in 1/10000ths of a basis point so our fee tier will be 3000: 
```
This should be 10000    0.3 * 10000 = 3000